BlobTierAnalysisTool/BlobTierAnalysisTool.NetCore/Helpers/BlobStorageHelper.cs

```diff
@@ -242,22 +201,23 @@ public static string StorageAccountName
         /// <param name="blob"></param>
         /// <param name="filterCriteria"></param>
         /// <returns></returns>
-        private static bool DoesBlobMatchFilterCriteria(CloudBlockBlob blob, Models.FilterCriteria filterCriteria)
+        private static bool DoesBlobMatchFilterCriteria(BlobClient blob, Models.FilterCriteria filterCriteria)
         {
-            if (blob.Properties.StandardBlobTier == StandardBlobTier.Archive) return false;
+            BlobProperties blobProperties = blob.GetProperties().Value;
+            if (blobProperties.AccessTier == AccessTier.Archive) return false;
             var dateTimeFrom = filterCriteria.LastModifiedDateFrom ?? DateTime.MinValue;
             var dateTimeTo = filterCriteria.LastModifiedDateTo ?? DateTime.MaxValue;
             var minBlobSize = filterCriteria.MinBlobSize;
             bool isDateTimeCheckPassed = false;
-            if (blob.Properties.LastModified.HasValue)
+            if (blobProperties.LastModified != null)
```

Comment:
LastModified is a required struct so it won't ever be null. I think comparing against `default` should give us the behavior we want.